src/libs/CurrencyUtils.ts

```diff
@@ -110,11 +111,20 @@ function convertToFrontendAmountAsString(amountAsInt: number | null | undefined)
  * @param amountInCents – should be an integer. Anything after a decimal place will be dropped.
  * @param currency - IOU currency
  */
-function convertToDisplayString(amountInCents = 0, currency: string = CONST.CURRENCY.USD): string {
+function convertToDisplayString(amountInCents = 0, currency: keyof CurrencyList | undefined = CONST.CURRENCY.USD): string {
     const convertedAmount = convertToFrontendAmountAsInteger(amountInCents);
+    /**
+     * Handle currency empty which can break the application.
+     * We're doing validation here so that we never send the value which will break the application.
+     * More: https://github.com/Expensify/App/issues/43004
+     */
+    let validatedCurrency = currency;
+    if (!currency || currency.length === 0 || currency.length > 3) {
+        validatedCurrency = CONST.CURRENCY.USD;
+    }
```

Comment:
I think we can simplify this condition to `!currency` and this will check for `undefined` or `''`
```javascript
let currencyWithFallback = currency;
if (!currency) {
    currencyWithFallback = CONST.CURRENCY.USD;
}
```
Also, I think it is self-explanatory

Reply:
Yea, but my idea is to make it explain the value shouldn't be empty string as well.

Reply:
> Yea, but my idea is to make it explain the value shouldn't be empty string as well.

Ok, we can simplify the comment, maybe like this `// Fallback currency to USD if it empty string or undefined`

Reply:
Done, updated.!

Reply:
@b4s36t4 I think this syntax is better
```javascript
let currencyWithFallback = currency;
if (!currency) {
    currencyWithFallback = CONST.CURRENCY.USD;
}
```